herddb-core/src/main/java/herddb/codec/RecordSerializer.java

```diff
@@ -120,6 +143,50 @@ public static int deserializeCompare(byte[] data, int type, Object cvalue) {
                 throw new IllegalArgumentException("bad column type " + type);
         }
     }
+    
+    public static int deserializeCompare(Bytes data, int type, Object cvalue) {
+        switch (type) {
+            case ColumnTypes.BYTEARRAY:
+                //TODO: not perform copy
```

Comment:
we do not have any special code path for columns of type byte[], we can work on this topic in the future if needed